src/app/components/kudos-add-new-idea/kudos-add-new-idea.component.js

```diff
@@ -37,10 +39,10 @@
                 showAddIdeaFormErrorMessage("Please enter idea");
                 return false;
             } else if ($scope.author.length > 30) {
```

Comment:
Why not just have `maxlength` attributes on the HTML of the input? This would remove the need for any if this logic in the controller